server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/projection/cassandra/EnqueuedMailsDao.java

```diff
@@ -136,76 +239,35 @@ private PreparedStatement prepareSelectFrom(Session session) {
-            .name(name)
-            .sender(sender)
-            .recipients(recipients)
-            .lastUpdated(lastUpdated)
-            .errorMessage(errorMessage)
-            .remoteHost(remoteHost)
-            .remoteAddr(remoteAddr)
-            .state(state)
-            .addAllHeadersForRecipients(perRecipientHeaders)
-            .attributes(toAttributes(rawAttributes))
-            .build();
-
-        ManageableMailQueue.MailQueueItemView mailQueueItemView = new ManageableMailQueue.MailQueueItemView(mail);
-        return new CassandraMailQueueProjectionItem(queueName, MailKey.of(name), mailQueueItemView);
-    }
+    private Stream<BucketAndSlice> bucketAndSliceStream(Date timeRangeStartFrom) {
+        Stream<Long> sliceTimestampStream = timeSliceStream(timeRangeStartFrom);
+        Stream<Integer> bucketStream = IntStream.range(0, projectionConfiguration.getBucketCount()).boxed();
 
-    private Map<String, Serializable> toAttributes(Map<String, ByteBuffer> rowAttributes) {
-        return rowAttributes.entrySet()
-            .stream()
-            .map(entry -> Pair.of(entry.getKey(), fromByteBuffer(entry.getValue())))
-            .collect(Guavate.toImmutableMap(Pair::getLeft, Pair::getRight));
+        return bucketStream
+            .flatMap(bucketId -> sliceTimestampStream.map(sliceTime -> BucketAndSlice.of(bucketId, sliceTime)));
     }
 
-    private Serializable fromByteBuffer(ByteBuffer byteBuffer) {
-        try {
-            byte[] data = new byte[byteBuffer.remaining()];
-            byteBuffer.get(data);
-            ObjectInputStream objectInputStream = new ObjectInputStream(new ByteArrayInputStream(data));
-            return (Serializable) objectInputStream.readObject();
-        } catch (IOException | ClassNotFoundException e) {
-            throw new RuntimeException(e);
-        }
+    private Stream<Long> timeSliceStream(Date timeRangeStartFrom) {
+        return sliceIdStream(timeRangeStartFrom)
+            .map(slicePace -> slicePace * projectionConfiguration.getSliceWindowSize())
+            .map(slicePaceInSecond -> slicePaceInSecond * 1000)
+            .boxed();
     }
 
-    private PerRecipientHeaders fromHeaderMap(Map<String, UDTValue> rawMap) {
-        PerRecipientHeaders result = new PerRecipientHeaders();
-
-        rawMap.forEach((key, value) -> result.addHeaderForRecipient(PerRecipientHeaders.Header.builder()
-                .name(value.getString(HEADER_NAME))
-                .value(value.getString(HEADER_VALUE))
-                .build(),
-            toMailAddress(key)));
-        return result;
-    }
+    private LongStream sliceIdStream(Date timeRangeStartFrom) {
+        Integer sliceWindowSize = projectionConfiguration.getSliceWindowSize();
 
-    private MailAddress toMailAddress(String rawValue) {
-        try {
-            return new MailAddress(rawValue);
-        } catch (AddressException e) {
-            throw new RuntimeException(e);
+        Date now = new Date();
```

Comment:
seriously, you should not write date handling code like that, java.time is here for you